intel_extension_for_transformers/llm/runtime/graph/models/model_utils/model_files.h

```diff
@@ -661,8 +780,14 @@ struct model_model_loader {
       file.seek(shard.file_off, SEEK_SET);
       tmp_buf.resize(lt.size * lt.world_size);
       file.read_raw(tmp_buf.addr, lt.size * lt.world_size);
-      // only copy part of weight form the tmp_buf of origin file
-      memcpy(lt.data, tmp_buf.addr + lt.rank * lt.size, lt.size);
+      size_t num_rows = lt.ne.size() == 1 ? 1 : lt.ne.at(1);
+      if (lt.type == NE_TYPE_JBLAS) {
+        void* dst_data = (void*)lt.data;
+        jblas_split_weight(tmp_buf.addr, &dst_data, lt.ne.at(0), num_rows, lt.rank, 0);
```

Comment:
Why both `TP_1D_ROW` and `TP_1D_COLUMN` call `jblas_split_weight(tmp_buf.addr, &dst_data, lt.ne.at(0), lt.ne.at(1), lt.rank, 0);`? I assume that they should be slightly different.